### PR TITLE
Python and dependency upgrade

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Ref https://trello.com/c/8QtXeaAO/3043-by-the-end-of-each-month-we-will-have-upgraded-one-project and https://trello.com/c/PcdXrDF5/3119-ci-audit-to-make-sure-its-consistent-throughout

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,7 +61,7 @@ jobs:
 
   sam_build:
     docker:
-      - image: amazon/aws-sam-cli-build-image-python3.10:latest
+      - image: amazon/aws-sam-cli-build-image-python3.8:latest
```
Comment (VirginiaDooley) [inline .circleci/config.yml:64]: Flagging this change in case it might cause deployment issues. Increasing to 3.10 caused a build error.

Comment (symroe) [reply .circleci/config.yml]: Have a read of this fun issue https://github.com/aws/aws-lambda-base-images/issues/31

Including my favourite comment:

> If your release cycle of Lambda containers with new Python versions takes longer than that of completely new Python versions, I'd say, there's something wrong with your release process.

tl;dr: AWS Lambda doesn't support 2.10 yet :(

**Comment by VirginiaDooley** [comment]
[CI Audit to make sure it's consistent throughout](https://trello.com/c/PcdXrDF5/3119-ci-audit-to-make-sure-its-consistent-throughout)

File: .circleci/config.yml (hunk 1 above)
Comment (VirginiaDooley) [reply .circleci/config.yml]: Oh yes, I remember you mentioning this in Slack. So I suppose we hold off on upgrading?

Comment (VirginiaDooley) [reply .circleci/config.yml:66]: I've [added comment for aws-lambda-base-images](https://github.com/DemocracyClub/Website/pull/628/commits/def0af5490e717e3dc14135e0c1f385e932b6f85)

**Comment by symroe** [comment]
I think this is going to be better if we use python 3.8 everywhere rather than just in the Lambda...I'm thinking we might have some 3.10 features that pass tests but then fail on Lambda. 

Can you revert the 3.10 update, back to 3.8. Everything else is :+1:

**Comment by VirginiaDooley** [comment]
> I think this is going to be better if we use python 3.8 everywhere rather than just in the Lambda...I'm thinking we might have some 3.10 features that pass tests but then fail on Lambda.
> 
> Can you revert the 3.10 update, back to 3.8. Everything else is 👍

That's done and updated here 57fc958b61ea978eaefc2d246de75bddb08d8c66
